Makefile

```diff
@@ -191,3 +176,24 @@ endef
 define local_image_exists
 $(shell docker images -q $1 2> /dev/null)
 endef
+
+# arguments:
+# $1 - release image repo, e.g. mesosphere/spark
+# $2 - release image tag, e.g spark-2.4.3-hadoop-2.9-k8s
+# $3 - dev image full name, e.g mesosphere/spark-dev:ab36f1f3691a8be2050f3acb559c34e3e8e5d66e
+define tag_and_push_image
+	$(eval RELEASE_IMAGE_FULL_NAME=$(1):$(2))
+	# check, if specified image already exists to prevent overwrites
+	if [[ -z "$(call remote_image_exists,$(1),$(2))" ]]; then
+		# check, if dev image is present in local docker registry
+		[[ -z "$(call local_image_exists,$(3))" ]] && docker pull $(3)
+		docker tag $(3) $(RELEASE_IMAGE_FULL_NAME)
+		echo "Pushing image \"$(RELEASE_IMAGE_FULL_NAME)\""
+		docker push $(RELEASE_IMAGE_FULL_NAME)
+	else
+		echo "Warning: image \"$(RELEASE_IMAGE_FULL_NAME)\" already exists, skipping overwrite."
```

Comment:
I think it makes sense to fail this target with `$(error image $(RELEASE_IMAGE_FULL_NAME) already exists, will not proceed with overwrite)` to indicate that the release job didn't publish any new image and some action is needed.